tfm_ram_report does not work with sdk-ng 0.15.0

**Describe the bug**
The `tfm_ram_report` target does not work with sdk-ng 0.15.0, output is as follows:
```
ninja tfm_ram_report
[0/4] Performing build step for 'tfm'
ninja: no work to do.
[1/2] cd /tmp/aa/zephyr/samples/tfm_integration/tfm_secure_partit...ntegration/tfm_secure_partition/_AA --workspace=/tmp/aa -d 99 ram
/tmp/aa/zephyr/samples/tfm_integration/tfm_secure_partition/_AA/tfm/bin/tfm_s.elf
Traceback (most recent call last):
  File "/tmp/aa/zephyr/scripts/footprint/size_report", line 804, in <module>
    main()
  File "/tmp/aa/zephyr/scripts/footprint/size_report", line 780, in main
    do_simple_name_matching(elf, symbol_dict, processed)
  File "/tmp/aa/zephyr/scripts/footprint/size_report", line 269, in do_simple_name_matching
    lineprog = dwarfinfo.line_program_for_CU(compile_unit)
  File "/home/jamie/.local/lib/python3.10/site-packages/elftools/dwarf/dwarfinfo.py", line 241, in line_program_for_CU
    top_DIE = CU.get_top_DIE()
  File "/home/jamie/.local/lib/python3.10/site-packages/elftools/dwarf/compileunit.py", line 92, in get_top_DIE
    top = DIE(
  File "/home/jamie/.local/lib/python3.10/site-packages/elftools/dwarf/die.py", line 94, in __init__
    self._parse_DIE()
  File "/home/jamie/.local/lib/python3.10/site-packages/elftools/dwarf/die.py", line 234, in _parse_DIE
    abbrev_decl = self.cu.get_abbrev_table().get_abbrev(self.abbrev_code)
  File "/home/jamie/.local/lib/python3.10/site-packages/elftools/dwarf/abbrevtable.py", line 36, in get_abbrev
    return self._abbrev_map[code]
KeyError: 17
FAILED: zephyr/cmake/reports/CMakeFiles/tfm_ram_report /tmp/aa/zephyr/samples/tfm_integration/tfm_secure_partition/_AA/zephyr/cmake/reports/CMakeFiles/tfm_ram_report 
cd /tmp/aa/zephyr/samples/tfm_integration/tfm_secure_partition/_AA && /usr/bin/python3.10 /tmp/aa/zephyr/scripts/footprint/size_report -k /tmp/aa/zephyr/samples/tfm_integration/tfm_secure_partition/_AA/tfm/bin/tfm_s.elf -z /tmp/aa/zephyr -o /tmp/aa/zephyr/samples/tfm_integration/tfm_secure_partition/_AA --workspace=/tmp/aa -d 99 ram
ninja: build stopped: subcommand failed.
```

Likely caused by the tfm build outputting a dwarfv5 elf instead of dwarfv4

**To Reproduce**
Build `samples/tfm_integration/tfm_secure_partition` for `nrf5340dk_nrf5340_cpuapp_ns` then run `ninja tfm_ram_report`

**Expected behavior**
The command to work

**Environment (please complete the following information):**
 - OS: Linux
 - Toolchain: Zephyr SDK 0.15.0
 - Commit SHA or Version used: b96453f653b674629c617bdbc810c3d9f7916ef4

--- a/west.yml
+++ b/west.yml
@@ -235,7 +235,7 @@ manifest:
       groups:
         - debug
     - name: trusted-firmware-m
-      revision: 4c984817d861512edaeca27b83308e9b9915a98a
+      revision: 231235f26f5295ac4faf8c5617dbb9779869d821
       path: modules/tee/tf-m/trusted-firmware-m
       groups:
         - tee

